### PR TITLE
PYTHON-3375 Added docstrings to DatetimeConversionOpts

### PR DESCRIPTION
**Pull request by bcwarner**
Added short docstrings to each of the `DatetimeConversionOpts`.

### CODE REVIEW

File: bson/codec_options.py [hunk 1]
```diff
@@ -201,9 +201,33 @@ def __eq__(self, other: Any) -> Any:
 
 class DatetimeConversionOpts(enum.IntEnum):
     DATETIME = 1
+    """Decode a BSON UTC datetime as a :class:`datetime.datetime`.
+
+    BSON UTC datetimes that cannot be represented as a
+    :class:`~datetime.datetime` will raise an :class:`~builtins.OverflowError`
```
Comment (ShaneHarvey) [inline bson/codec_options.py:207]: Is it always an OverflowError? I recall sometimes it's a ValueError.

Comment (bcwarner) [reply bson/codec_options.py:207]: Yeah, I think it can looking at the source for `datetime.datetime`

**Comment by bcwarner** [comment]
Fails on an unrelated CSOT test

File: bson/codec_options.py [hunk 2]
```diff
@@ -201,9 +201,34 @@ def __eq__(self, other: Any) -> Any:
...
+    or a :class:`~builtins.ValueError`
```
Comment (ShaneHarvey) [inline bson/codec_options.py:208]: missing punctuation.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**